### PR TITLE
Fix k8s pod labels tier in metadata

### PR DESCRIPTION
**Pull request by ChrsMark**
Closes https://github.com/elastic/beats/issues/16464

Description of bug investigation: https://github.com/elastic/beats/issues/16464#issuecomment-589585357

cc: @exekias

### CODE REVIEW

File: libbeat/common/kubernetes/metadata/pod.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ func (p *pod) Generate(obj kubernetes.Resource, opts ...FieldOptions) common.Map
 	}
 
 	out := p.resource.Generate("pod", obj, opts...)
+	// TODO: remove this call when moving to 8.0
```
Comment (mtojek) [inline libbeat/common/kubernetes/metadata/pod.go]: I thinks it's fair to create an issue for removing this TODO ;)

File: libbeat/common/kubernetes/metadata/pod.go [hunk 2]
```diff
@@ -89,3 +91,14 @@ func (p *pod) GenerateFromName(name string, opts ...FieldOptions) common.MapStr
 
 	return nil
 }
+
+func (p *pod) fixLabels(in common.MapStr) common.MapStr {
```
Comment (mtojek) [inline libbeat/common/kubernetes/metadata/pod.go]: Hmm.. "fix" says nothing about the behavior. Could you rename this method to sth less vague? `movePodLabels`, `extractPodLabel`, etc.

File: libbeat/common/kubernetes/metadata/pod.go (hunk 1 above)
Comment (ChrsMark) [reply libbeat/common/kubernetes/metadata/pod.go]: https://github.com/elastic/beats/issues/16483

**Comment by exekias** [comment]
LGTM, Let's move this one forward, it seems some tests are failing, we need to update some tests cases

File: libbeat/common/kubernetes/metadata/pod.go (hunk 1 above)
Comment (exekias) [reply libbeat/common/kubernetes/metadata/pod.go]: +1, that said. I'm not quite convinced this is the way to go. Let's move this fix forward and discuss future steps in the opened issue

**Comment by ChrsMark** [comment]
Test cases fixed. Also added a changelog.

**Pull request merged**